PowerPointLabs/Test/Util/SlideUtil.cs

```diff
@@ -20,25 +20,88 @@ public static void IsSameShape(Shape expShape, Shape actualShape)
+                Assert.IsTrue(expText.Font.Bold == actualText.Font.Bold,
+                    "different bold style for {0}. exp:{1}, actual:{2}", expShape.Name, expText.Font.Bold, expText.Font.Bold);
+                Assert.IsTrue(expText.Font.Underline == actualText.Font.Underline,
+                    "different underline style for {0}. exp:{1}, actual:{2}", expShape.Name, expText.Font.Underline, expText.Font.Underline);
+                Assert.IsTrue(expText.Font.Italic == actualText.Font.Italic,
+                    "different italic style for {0}. exp:{1}, actual:{2}", expShape.Name, expText.Font.Italic, expText.Font.Italic);
+                Assert.IsTrue(expText.Font.Size == actualText.Font.Size,
+                    "different text size for {0}. exp:{1}, actual:{2}", expShape.Name, expText.Font.Size, expText.Font.Size);
+            }
         }
 
         public static void IsSameZOrderPosition(Shape expShape, Shape actualShape)
         {
             Assert.IsTrue(expShape.ZOrderPosition == actualShape.ZOrderPosition,
-                "different shape z order. exp:{0}, actual:{1}", expShape.ZOrderPosition, actualShape.ZOrderPosition);
+                "different shape z order for {0}. exp:{1}, actual:{2}", expShape.Name, expShape.ZOrderPosition, actualShape.ZOrderPosition);
+        }
+
+        public static void IsSameShapes(Slide expSlide, Slide actualSlide)
+        {
+            var expShapes = expSlide.Shapes;
+            var actualShapes = actualSlide.Shapes;
+            Assert.IsTrue(expShapes.Count == actualShapes.Count,
+                "different number of shapes on slide. exp:{0}, actual:{1}", expShapes.Count, actualShapes.Count);
+
+            for (int i = 1; i <= expShapes.Count; i++)
+            {
+                var expShape = expShapes[i];
+                bool isMatchingShapeFound = false;
+                for (int j = 1; j <= actualShapes.Count; j++)
+                {
+                    var actualShape = actualShapes[j];
+                    if (expShape.Name == actualShape.Name)
+                    {
+                        isMatchingShapeFound = true;
+                        if (expShapes[i].Type == Microsoft.Office.Core.MsoShapeType.msoGroup)
+                        {
+                            Assert.IsTrue(expShape.Type == actualShape.Type,
+                                "different shape type for {0}. exp:{1}, actual:{2}", expShape.Name, expShape.Type, actualShape.Type);
+                            Assert.IsTrue(expShape.GroupItems.Count == actualShape.GroupItems.Count,
+                                "different number of shapes in group for {0}. exp:{1}, actual:{2}",
+                                expShape.Name, expShape.GroupItems.Count, actualShape.GroupItems.Count);
+
+                            for (int k = 1; k <= expShape.GroupItems.Count; k++)
+                            {
+                                IsSameShape(expShape.GroupItems[k], actualShape.GroupItems[k]);
```

Comment:
Oh. I did not think know that you could have a group within a group in PowerPoint. I will make the changes